src/middlewares/item-authorization.js

```diff
@@ -0,0 +1,12 @@
+/* eslint-disable no-underscore-dangle */
+const Item = require('../models/item');
+
+module.exports = async (req, res, next) => {
+  const itemId = req.params.id;
+  const item = await Item.findById(itemId);
+  if (item && item.owner._id.toString() === req.user._id) {
```

Comment:
Since the `item.owner` is the reference to the user you don't need to use `._id`, 
`item.owner.toString()` is enough.